### PR TITLE
Remove dependency on bash

### LINKED ISSUE #1: Use `sh` shell
Hey.

Currently this uses `bash` shell: https://github.com/liskin/gh-workflow-keepalive/blob/main/action.yml#L18

When running on GitHub hosted runners, it works okay as they contain both `bash` and `gh` CLI. When you try to run the same action on self-hosted runners, they may not have `gh` CLI set up. To counter this, I'm running this action from a container image that has `gh`, but doesn't have `bash` installed: https://github.com/maniator/gh

Of course I could just install `bash` into that image, but I believe a better solution would be to just drop the dependency on `bash`. It's only needed for the first line of the `run`, and although I'm unsure why this condition is necessary, instead of evaluating this result with `bash`, it could be replaced with `if` checks from GH itself: 
`if: ${{ startsWith(github.workflow_ref, format('{0}/.github/workflows/', github.repository)) }}`

Is this something you'll accept as a PR?

### PR DESCRIPTION
**Pull request by oprypkhantc**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -15,9 +15,9 @@ runs:
     - name: Re-enable workflow
       env:
         GITHUB_TOKEN: ${{ inputs.GITHUB_TOKEN }}
-      shell: bash
+      shell: sh
       run: |
-        [[ "${GITHUB_WORKFLOW_REF:?}" == "${GITHUB_REPOSITORY:?}"/.github/workflows/*.y?(a)ml@* ]]
+        expr "${GITHUB_WORKFLOW_REF:?}" : "${GITHUB_REPOSITORY:?}\/\.github\/workflows\/.*\.ya\?ml\@.*" > /dev/null
```
Comment (liskin) [inline action.yml]: `expr` is not a shell builtin, and we're not escaping the contents of `$GITHUB_REPOSITORY`, so any special regex characters in it keep their special meaning.

I think a better way to translate the `[[` bashism to posix sh is using `case`. Something like:
```sh
case "${GITHUB_WORKFLOW_REF:?}" in
  "${GITHUB_REPOSITORY:?}"/.github/workflows/*.y*ml@*) ;;
  *) false ;;
esac
```

It isn't particularly nice but that's not exactly surprising of POSIX sh :-)

Comment (oprypkhantc) [reply action.yml]: Yeah I missed the escaping part :) Thank you. Pushed the changes.

**Review by liskin — APPROVED** [review]
LGTM

Can you please confirm that you've tested this and it works in all your usecases so I don't have to? :-D

**Comment by oprypkhantc** [comment]
I've tested a few "negative" cases locally and both positive and negative tests on a project of mine, seems to be working as expected :)

https://github.com/tenantcloud/php-standard/actions/runs/8897445975/job/24432730834
https://github.com/tenantcloud/php-standard/actions/runs/8897445975/job/24432349815

**Comment by liskin** [comment]
Awesome, thanks a lot for the contribution! :tada: :beers:

**Pull request merged**